src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -1349,44 +1349,38 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
             break;
 
             case NI_AdvSimd_Insert:
+            {
                 assert(isRMW);
 
                 GetEmitter()->emitIns_Mov(INS_mov, emitTypeSize(node), targetReg, op1Reg, /* canSkip */ true);
 
-                if (intrin.op3->isContainedFltOrDblImmed())
-                {
-                    assert(intrin.op2->isContainedIntOrIImmed());
-                    assert(intrin.op2->AsIntCon()->gtIconVal == 0);
+                // fmov (scalar) zeros the upper bits and is not safe to use
+                assert(!intrin.op3->isContainedFltOrDblImmed());
```

Comment:
GitHub is displaying the diff a bit weirdly. This is just removing the `if` and fixing the indentation of the `else` block now that its the only code path.